### PR TITLE
Newsletter UI improvements (full route + embedded form)

### LINKED ISSUE #299: Create Newsletter Opt-in embedded in the page

### PR DESCRIPTION
**Pull request by allozaur**
Close #299
Close tradingstrategy-ai/design-system#18
## Demo

https://user-images.githubusercontent.com/37627284/211072413-92ec1f58-671f-45f0-a6fb-957150d174c1.mp4

### CODE REVIEW

**Comment by miohtama** [comment]
This is good progress. However we should still do some improvement on the content and stlyin

Can you make it more boxed like that it is separated from the content?

Like here

image

image


image

image

The initial call to action and catchphrase should be brainstormed with Lawrence and have some more meat on the bones than "Sign up to newsletter"

We can have it in some similar fashion as we have About page boxes

image

**Comment by allozaur** [comment]
@miohtama here is an update for the design.

Now we can just embed only the Subscribe Form component like here:

Zrzut ekranu 2023-01-20 o 21 45 00

or if we want to have a more content and media rich opt in banner we can have it like this:

### Video

https://user-images.githubusercontent.com/37627284/213802746-2c97e11a-0029-4682-b412-b05ca6f8ee0d.mov

### Screenshots




Zrzut ekranu z 2023-01-20 o 21 53 08

Zrzut ekranu z 2023-01-20 o 21 53 21

Zrzut ekranu z 2023-01-20 o 21 53 29

Zrzut ekranu z 2023-01-20 o 21 53 33

**Comment by allozaur** [comment]
@miohtama here is an update for the design.

Now we can just embed only the Subscribe Form component like here:

Zrzut ekranu 2023-01-20 o 21 45 00

or if we want to have a more content and media rich opt in banner we can have it like this:

### Video

https://user-images.githubusercontent.com/37627284/213802746-2c97e11a-0029-4682-b412-b05ca6f8ee0d.mov

### Screenshots




Zrzut ekranu z 2023-01-20 o 21 53 08

Zrzut ekranu z 2023-01-20 o 21 53 21

Zrzut ekranu z 2023-01-20 o 21 53 29

Zrzut ekranu z 2023-01-20 o 21 53 33

**Comment by miohtama** [comment]
There seem to be a misunderstanding here.

- Strategies coming soon is one section
- Newsletter is another section

Now these sections are  merged.

Please give a basic newsletter subscription banner that is correctly separated from the rest of the page border like any newsletter subscription form I gave in an example.

**Comment by allozaur** [comment]
@miohtama something like this?

Zrzut ekranu 2023-01-23 o 13 21 05

Zrzut ekranu 2023-01-23 o 13 20 11

**Comment by allozaur** [comment]
The `/newsletter` page looks like this:


https://user-images.githubusercontent.com/37627284/214040550-1d77ca7b-1c97-40db-a9ff-f9472d5cd25f.mov

File: src/routes/qr/+page.svelte
Comment (kenkunz) [inline src/routes/qr/+page.svelte:86]: Page looks broken w/out this CSS.

Fix – use the new `ContentCardTemplate` / `ContentCardSection` components.

File: src/routes/qr/+page.ts
Comment (kenkunz) [inline src/routes/qr/+page.ts:46]: I prefer keeping this as explicit `ContentCard` props in `+page.svelte`.

File: src/lib/components/SubscribeForm.svelte
Comment (kenkunz) [inline src/lib/components/SubscribeForm.svelte:9]: `<SubscribeForm />`

File: src/lib/components/SubscribeForm.svelte
Comment (kenkunz) [inline src/lib/components/SubscribeForm.svelte:5]: Docs need updating (no longer accurate).

File: src/lib/components/SubscribeForm.svelte
Comment (kenkunz) [inline src/lib/components/SubscribeForm.svelte:17]: no longer used in this component

File: src/routes/newsletter/subscribe/+page.server.ts
Comment (kenkunz) [inline src/routes/newsletter/subscribe/+page.server.ts:2]: Accidental? This should be reverted to `./`

**Review by kenkunz — CHANGES_REQUESTED** [review]
See file-specific comments (all of them relatively minor).

**Comment by kenkunz** [comment]
@allozaur, @miohtama – my 2¢ on the UX:

### Embedded form 👍

The embedded form (e.g., on the home page) is great – definitely an improvement over the modal dialog for that context.

### Full page form vs. modal 👎

That said, I believe the **modal dialog** provides an objectively better user experience than linking to a **separate page** in contexts where it doesn't make sense to have an embedded form (footer, social links, etc):
1. dialog requires 1 less click since the form field is focused when you open the dialog
2. dialog keeps the user in their original context so they don't need to navigate back via back button or manually
3. a full page load is just as "modal" (in the sense of preventing other user interactions) as a modal dialog

Both of these _could_ be addressed / improved in the **full page** version, but…
1. auto-focusing a form field on page-load is generally frowned upon for accessibility reasons
2. adding a "return-to" link on the form page feels like a work-around

@miohtama I know u felt passionately about getting rid of the modal – just wanted u to see the two experiences side-by-side before making a final call to go with the full page experience:

https://user-images.githubusercontent.com/35901/214178341-d9b1d376-9fb8-4541-b556-3b743531d69d.mov

**Comment by allozaur** [comment]
@miohtama visual updates are ready for review. @kenkunz i will also address your code review comments :)

https://user-images.githubusercontent.com/37627284/214259740-9d2099f5-400e-4ccf-a2d6-e822f9bcaa68.mov

**Comment by allozaur** [comment]
> ### Full page form vs. modal 👎

@kenkunz i agree with your opinion on that. I think that both full page and the modal dialog have their own user stories in which they shine. @miohtama the final call is yours of course

**Comment by miohtama** [comment]
Let's go with full page always when it is an option

**Comment by miohtama** [comment]
Approved

File: src/lib/components/Footer.svelte
Comment (allozaur) [reply src/lib/components/Footer.svelte:40]: > Is --footer-height used anywhere outside Footer.svelte? If not, let's keep its CSS scope local to this component.

Will do!

File: src/lib/components/css/reset.css
Comment (allozaur) [reply src/lib/components/css/reset.css]: Okay, maybe then it could be ussed just for defined context or params for routes? For now i just moved this to `src/routes/+page.svelte`. It's a little bit weird though that this quirky scrolling issue arises after globally applying this rule...

For now i just moved this to Home page component — 86626e177de562e7f798562ff2a602f7977b21f1

File: src/routes/qr/+page.svelte
Comment (allozaur) [reply src/routes/qr/+page.svelte:86]: yeah, actually i had it somewhere... maybe it got lost after solving the merge conflicts. Regardless — it will not take long

File: src/routes/newsletter/subscribe/+page.server.ts
Comment (allozaur) [reply src/routes/newsletter/subscribe/+page.server.ts:2]: yep, accidental

File: src/lib/components/SubscribeForm.svelte
Comment (allozaur) [reply src/lib/components/SubscribeForm.svelte]: updated

File: src/lib/components/SubscribeForm.svelte
Comment (allozaur) [reply src/lib/components/SubscribeForm.svelte]: IDE shows me that just line `17` is not needed so i removed only this one

**Comment by allozaur** [comment]
@kenkunz post CR code changes are applied. Please review it :)

File: src/routes/+page.svelte
Comment (kenkunz) [inline src/routes/+page.svelte:75]: FYI – this still "leaks" the `scroll-behavior` onto the rest of the app. Expect any un-scoped `global` CSS in any component to affect your entire app.

Since `scroll-behavior` must be attached to the root element, scoping it is it bit tricker. Here's an option that works well – keeps the relevant CSS co-located in the same component, prevents "leakage":

```css
	:global html:has(#home-newsletter) {
		scroll-behavior: smooth;
	}
```

No need to change this – I have it in a local commit and will push shortly.

**Review by kenkunz — COMMENTED** [review]
Great progress! I will push a couple minor tweaks as well as fix failing tests and then merge.

File: src/routes/+page.svelte
Comment (allozaur) [reply src/routes/+page.svelte:75]: okay! although `:has` tag is only supported by Safari/WebKit...

Or does PostCSS Preset Env make this work??

Comment (kenkunz) [reply src/routes/+page.svelte:75]: > Or does PostCSS Preset Env make this work??

indeed 😄

Anything stage 2 or above here should work:
https://cssdb.org/

That said… I've seen some inconsistent/buggy behavior with `:has` pseudo – so use with caution and test it!

**Review by kenkunz — APPROVED** [review]
Should be good-to-go!

**Pull request merged**
